pylabrobot/tilting/tilter.py

```diff
@@ -1,89 +1,213 @@
 
-      # Convert angle to radians.
-      theta = math.radians(angle)
+    new_x = x_prime + (self._hinge_coordinate.x + self.get_absolute_location("l", "f", "b").x)
+    new_z = z_prime + (self._hinge_coordinate.z + self.get_absolute_location("l", "f", "b").z)
 
-      # Compute the current location of the well. Use the rotation anchor as the origin.
-      x = self.get_plate().get_size_x() - well.location.x
-      z = well.location.z
-      h = math.sqrt(x**2 + z**2) # hypotenuse (dist from anchor to well)
+    return Coordinate(new_x, absolute_coordinate.y, new_z)
+
+  def return_well_drain_offsets(self, position: int, absolute_angle: int = None):
+    """
+    Return the edge offset for the wells in the plate at a given position, rotated at a given absolute angle.
+
+    Args:
+      position (int): The position index of the plate.
+      absolute_angle (int, optional): The absolute angle to rotate the plate. Defaults to None.
+
+    Returns:
+      List[Coordinate]: A list of offsets for the wells in the plate.
+    """
 
-      d = 1 # offset of well from tilting plane (perpendicular to tilt axis)
+    if absolute_angle is None:
+      angle = self._absolute_angle if self._hinge_coordinate.x < self._size_x / 2 else -self._absolute_angle
+    else:
+      angle = absolute_angle if self._hinge_coordinate.x < self._size_x / 2 else -absolute_angle
 
-      # Compute the new location of the well after rotation.
-      x_prime = h * math.cos(theta) + d * math.cos(theta+(math.pi/2)) # x component of vector
-      z_prime = h * math.sin(theta) + d * math.sin(theta+(math.pi/2)) # z component of vector
+    _hinge_side = "l" if self._hinge_coordinate.x < self._size_x / 2 else "r"
 
-      well.location.x = self.get_plate().get_size_x() - x_prime
-      well.location.z = z_prime
+    offsets = []
+    for well in self.get_plate(position).children:
+      zero_absolute_well_drain_coordinate = well.get_absolute_location(_hinge_side, "c", "b")
+      print(zero_absolute_well_drain_coordinate)
+      rotated_absolute_well_drain_coordinate = self.rotate_coordinate_around_hinge(
+        zero_absolute_well_drain_coordinate, angle
+      )
+      print(rotated_absolute_well_drain_coordinate)
+      drain_offset = rotated_absolute_well_drain_coordinate - well.get_absolute_location("c", "c", "b")
+      offsets.append(drain_offset)
 
-    self._angle = angle
+    return offsets
 
   async def tilt(self, angle: int):
-    """ Tilt the plate contained in the tilt module by a given angle relative to the current angle.
+    """Tilt the plate contained in the tilt module by a given angle relative to the current angle.
 
     Args:
-      angle: The angle to rotate by, in degrees. Clockwise. 0 is horizontal.
+    angle: The angle to rotate by, in degrees. Clockwise. 0 is horizontal.
     """
 
-    await self.set_angle(self.angle + angle)
+    await self.set_angle(self._absolute_angle + angle)
```

Comment:
Note that there is a distinction between `angle` and `absolute_angle` now. `absolute_angle` is what is exposed to the end user, and is given a sign internally to become `angle`, which simplifies calculations.